doc/source/build_hints.rst

```diff
@@ -2014,6 +2020,11 @@ Python bindings options
     Whether Python bindings should be built. It is ON by default, but only
     effective if a Python installation is found.
 
+.. option:: SWIG_REGENERATE_PYTHON:BOOL=ON/OFF
+
+    Whether to refresh the generated SWIG Python bindings. It is OFF by default.
+    Setting it to ON is needed if modifying the SWIG interface files.
```

Comment:
On thing I've been unsure about -- is this something which should be run by a contributor prior to submitting a change, or is it something which should instead be run prior to release/as required on an ad-hoc basis?

I get a lot of noise when I regenerate the bindings automatically and usually resort to just manually updating the files (or just copy/pasting the actual useful changes). 

I think it would be good to clarify the expectations somewhere...

Reply:
> - is this something which should be run by a contributor prior to submitting a change

yes. Maybe at some point we will require SWIG to be able to build GDAL, but this is not yet the case.

> I get a lot of noise when I regenerate the bindings automatically 

you have to use SWIG 4.0.2.  I'm not sure how you regenerated the bindings. If that was from a "make generate" from swig/python (which will be eventually retired once we've removed autoconf support), then I wouldn't expect noise.

> I think it would be good to clarify the expectations somewhere...

the updated CONTRIBUTING.md normally gives the good hints. Do you see additions that would be useful ?